stp_zmq/zstack.py

```diff
@@ -774,25 +780,31 @@ def send(self, msg: Any, remoteName: str = None, ha=None):
                 return self.transmit(msg, remoteName)
 
     def transmit(self, msg, uid, timeout=None):
-        # Timeout is unused as of now
-        assert uid in self.remotes
-        socket = self.remotes[uid].socket
-        if socket:
-            msg = self.prepMsg(msg)
-            try:
-                # noinspection PyUnresolvedReferences
-                # socket.send(self.signedMsg(msg), flags=zmq.NOBLOCK)
-                socket.send(msg, flags=zmq.NOBLOCK)
-                logger.debug(
-                    '{} transmitting message {} to {}'.format(self, msg, uid))
-                return True
-            except zmq.Again as ex:
-                logger.debug('{} could not transmit message to {}'.format(self, uid))
-                return False
-        else:
-            logger.warning('{} has uninitialised socket for remote {}'.
-                        format(self, self.remotes[uid]))
+        remote = self.remotes.get(uid)
+        if remote is None:
+            logger.error("Remote {} does not exist!".format(uid))
+            return False
+        socket = remote.socket
+        if not socket:
+            logger.info('{} has uninitialised socket '
+                        'for remote {}'.format(uid))
             return False
+        try:
+            msg = self.prepMsg(msg)
+            # socket.send(self.signedMsg(msg), flags=zmq.NOBLOCK)
+            socket.send(msg, flags=zmq.NOBLOCK)
+            logger.info('{} transmitting message {} to {}'
+                        .format(self, msg, uid))
+            if not remote.isConnected:
+                logger.warning('Remote {} is not connected - '
```

Comment:
- Shouldn't we mention that tcp ports must be checked?
- Shouldn't we move this check before `socket.send`? Otherwise we will be thrown right into except statement, won't we?